src/script/interpreter.cpp

```diff
@@ -61,6 +61,75 @@ static inline void popstack(std::vector<valtype>& stack)
     stack.pop_back();
 }
 
+static inline void push4_le(std::vector<valtype>& stack, uint32_t v)
+{
+    auto v_le = htole32(v);
+    valtype vch(reinterpret_cast<unsigned char*>(&v_le), reinterpret_cast<unsigned char*>(&v_le) + 4);
+    stack.push_back(vch);
+}
+
+static inline void push8_le(std::vector<valtype>& stack, uint64_t v)
+{
+    auto v_le = htole64(v);
+    valtype vch(reinterpret_cast<unsigned char*>(&v_le), reinterpret_cast<unsigned char*>(&v_le) + 8);
+    stack.push_back(vch);
+}
+
+static inline void pushasset(std::vector<valtype>& stack, const CConfidentialAsset& asset)
+{
+    assert(!asset.IsNull());
+    valtype vchinpAsset;
+    vchinpAsset.insert(vchinpAsset.begin(), asset.vchCommitment.begin() + 1, asset.vchCommitment.begin() + 33);
+    valtype vchAssetPref;
+    vchAssetPref.insert(vchAssetPref.begin(), asset.vchCommitment.begin(), asset.vchCommitment.begin() + 1);
+    stack.push_back(vchinpAsset);
+    stack.push_back(vchAssetPref);
+}
+
+static inline void pushvalue(std::vector<valtype>& stack, const CConfidentialValue& value)
+{
+    valtype vchinpValue;
+    if (value.IsNull()) {
+        vchinpValue.insert(vchinpValue.begin(), 8, 0);
```

Comment:
pieter suggested that `assign` is more appropriate than `insert` in these cases where you cannot do constructor initialization.  But otherwise, where possible, you should replace your use of `insert` with constructor initialization.